### ISSUE TITLE
CallbackContract unit test fails when run with NET Native

### ISSUE THREAD

**Issue by roncain** [dotnet/wcf#110]
Using the experimental tools to run the GitHub tests against the NET Native toolchain, the CallbackContract unit test fails:

  --------------------------------
  Running Test: ContractDescriptionTest.Duplex_ContractDescription_Builds_From_ServiceContract
  Assert.Equal:Expected: [2]. Actual:[1]. 
  Xunit.AssertTestException: Assert.Equal: Expected: [2]. Actual:[1]. 

Early analysis shows the NET Native environment does not naturally allow reflection over the type declared in the CallbackContractg

**Comment by roncain** [comment]
The issue is that the NET Native analysis engine does not recognize that the type specified in ServiceContractAttribute.CallbackContract is used, and therefore its metadata is removed. Moreover, there is no existing rd.xml directive that can be applied to named parameters on attributes.  The "Method" directive used to specify the "set_CallbackContract" setter is potentially the right approach, but the toolchain does not detect that this setter method is called when attributes are just used declaratively.

I opened a separate TFS issue against the NET Native team requesting a fix for this (they are not in GitHub so it could only be TFS). I am leaving this GitHub  issue open to track progress of the other.

In the meantime to make progress, I have discovered a default.rd.xml (as VS gives every NET Native project now) placed in the unit test's folder will allow this test to succeed.
